Common/src/Pollster/TaskHandler.cs

```diff
@@ -214,13 +231,55 @@ public async Task<bool> AcquireTask(CancellationToken cancellationToken)
         return false;
       }
 
-      logger_.LogDebug("checking that the number of retries is not greater than the max retry number");
-      var acquireTask = await taskTable_.AcquireTask(messageHandler_.TaskId,
-                                                     cancellationToken)
-                                        .ConfigureAwait(false);
-      if (!acquireTask)
+      var ownerPodId = LocalIPv4.GetLocalIPv4Ethernet();
+
+      logger_.LogDebug("Trying to acquire task");
+      taskData_ = await taskTable_.AcquireTask(messageHandler_.TaskId,
+                                               ownerPodId,
+                                               cancellationToken)
+                                  .ConfigureAwait(false);
+
+      // we check if the task was acquired by this pod
+      if (taskData_.OwnerPodId != ownerPodId)
       {
-        messageHandler_.Status = QueueMessageStatus.Postponed;
+        // if the task is acquired by another pod, we check if the task is running on the other pod
+        var taskProcessingElsewhere = await taskProcessingChecker_.Check(taskData_.TaskId,
+                                                                         taskData_.OwnerPodId,
+                                                                         cancellationToken)
+                                                                  .ConfigureAwait(false);
+
+        logger_.LogInformation("Task {taskId} already acquired by {OtherOwnerPodId}, treating it {processing}",
+                               taskData_.TaskId,
+                               taskData_.OwnerPodId,
+                               taskProcessingElsewhere);
+
+        // if the task is not running on the other pod, we resubmit the task in the queue
+        if (!taskProcessingElsewhere)
+        {
+          taskData_ = await taskTable_.ReadTaskAsync(messageHandler_.TaskId,
+                                                     cancellationToken)
+                                      .ConfigureAwait(false);
+          if (taskData_.OwnerPodId != ownerPodId && taskData_.Status is TaskStatus.Processing or TaskStatus.Dispatched)
```

Comment:
Does not need to check for `taskData_.OwnerPodId != ownerPodId`, but would most likely need to check `taskData_.Status is not TaskStatus.Completed and not TaskStatus.Error`

Reply:
Or just test for completed (because the other test is done inn resubmit).